internal/module/currency/init.go

```diff
@@ -0,0 +1,37 @@
+package currency
+
+import (
+	"genesis-currency-api/internal/module/currency/api/handler"
+	"genesis-currency-api/internal/module/currency/service"
+	sharcurrdto "genesis-currency-api/internal/shared/dto/currency"
+	"github.com/gin-gonic/gin"
+)
+
+type Provider interface {
+	GetCurrencyRate() (*sharcurrdto.ResponseDTO, error)
+}
+
+type Service interface {
+	GetCurrencyRate() (sharcurrdto.ResponseDTO, error)
+	GetCachedCurrency() (sharcurrdto.CachedCurrency, error)
+	UpdateCurrencyRates() error
+}
+
+type Handler interface {
+	GetLatest(ctx *gin.Context)
+}
+
+type Module struct {
+	Service Service
+	Handler Handler
+}
```

Comment:
It's cool that you separate infrastructure components by DIP. This helps you to inject mocks objects for more general web tests without real repositories/services👍

Reply:
But for me, this is the only reason. This code is called from the infrastructure layer and can have explicit dependencies to reduce the abstraction of objects in the project and simplify it.